easybuild/easyconfigs/c/CrayCCE/CrayCCE-19.06-clang.eb

```diff
@@ -0,0 +1,27 @@
+easyblock = 'CrayToolchain'
+
+name = 'CrayCCE'
+version = '19.06-clang'
+
+homepage = 'https://pubs.cray.com/discover'
+description = """Toolchain using Cray compiler wrapper, using PrgEnv-cray
+(PE release: June 2019).\n"""
+
+toolchain = {'name': 'dummy', 'version': ''}
```

Comment:
If I use version `''` instead of `dummy` as [in the other versions](https://github.com/eth-cscs/production/blob/a6c450c1ab856213425bdb30520de79885b17093/easybuild/easyconfigs/c/CrayGNU/CrayGNU-19.03.eb#L12). I get the following error.
```
== FAILED: Installation ended unsuccessfully: build failed (first 300 chars): PrgEnv-gnu/6.0.5(76):ERROR:150: Module 'PrgEnv-gnu/6.0.5' conflicts with the currently loaded module(s) 'PrgEnv-cray/6.0.5'
```

Reply:
That's probably a clash with the modules that are loaded by default?

Are you only seeing this when installing the module for this toolchain, and not when installing something using this toolchain?

We take special measures in the module file generated for a `Cray*` toolchain w.r.t. `PrgEnv-*` (we include `unload` statements for all known `PrgEnv-*` modules to avoid this exact issue).

But that's not done when the toolchain module itself is being installed. The easy workaround for that is indeed to use `'dummy'` as a toolchain version, since then the `dependecies` don't get loaded (there's no reason to load them, since installing a toolchain doesn't involve any compilation).

Reply:
@hvictor I don't get this error when building `eb CrayCCE-19.06.eb`: I have tried on Piz Daint after loading 'cdt/19.06` using the amended `craytoolchain.py` that I have used for Tsa and Arolla (see the file in my the branch `tsa` of my fork at https://github.com/lucamar/production/blob/tsa/easybuild/easyblocks/craytoolchain.py).

I did the following on Piz Daint:
```
module load cdt/19.06
export EASYBUILD_PREFIX=/apps/daint/UES/sandbox/lucamar/easybuild
export EB_CUSTOM_REPOSITORY=/apps/common/UES/sandbox/lucamar/production/easybuild
module load daint-gpu
module load EasyBuild-custom/cscs
eb CrayCCE-19.06.eb
```

Reply:
@lucamar Your changes indeed work around the problem, why not submit a PR for that to the central `easybuild-easyblocks` repo?

Reply:
@boegel I will, it was already in my to-do-list, sorry for the delay...

Reply:
@lucamar, I do not get the error when installing `CrayCCE-19.06.eb` too. Even without your changes. The error I get when installing `CrayGNU-19.06.eb`. Sorry for pointing it out in the wrong file. And I still get the same errors when I try using your toolchain change.

Reply:
@boegel, yes. The issue is what you have described. Adding the `dummy` avoids the initial module load. Since we require some external modules, if they are not present, the build will fail. So, I think that, in this case, the `dummy` is not a workaround, but a requirement.

Reply:
Just to provide more info...
If I use @lucamar changes and add some `print`s. 
```
...
KNOWN_PRGENVS = ['PrgEnv-cray', 'PrgEnv-gnu', 'PrgEnv-intel', 'PrgEnv-pgi']
AVAIL_PRGENVS=[]
for prgenv in [prgenv for prgenv in KNOWN_PRGENVS]:
    print("-------------------")
    print(" Here, I am here")
    print("-------------------")

    try:
        subprocess.check_call("module unload %s &> /dev/null" % prgenv, shell=True)
        AVAIL_PRGENVS.append(prgenv)
    except subprocess.CalledProcessError:
        print("-------------------")
        print(" Here, I have failed")
        print("-------------------")

        continue
...
```

I get:
```
-------------------
 Here, I am here
-------------------
-------------------
 Here, I am here
-------------------
-------------------
 Here, I am here
-------------------
-------------------
 Here, I am here
-------------------
== temporary log file in case of crash /tmp/hvictor/gpu/eb-2lYitE/easybuild-J7Ahrs.log
== processing EasyBuild easyconfig /users/hvictor/EASYBUILD/production-new-pe-19.06/easybuild/easyconfigs/c/CrayGNU/CrayGNU-19.06.eb
== building and installing CrayGNU/19.06...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== FAILED: Installation ended unsuccessfully (build directory: /run/user/23962/easybuild/tmp/CrayGNU/19.06/dummy-): build failed (first 300 chars): PrgEnv-gnu/6.0.5(76):ERROR:150: Module 'PrgEnv-gnu/6.0.5' conflicts with the currently loaded module(s) 'PrgEnv-cray/6.0.5'
== Results of the build can be found in the log file(s) /tmp/hvictor/gpu/eb-2lYitE/easybuild-CrayGNU-19.06-20190628.113224.DFrTE.log
ERROR: Build of /users/hvictor/EASYBUILD/production-new-pe-19.06/easybuild/easyconfigs/c/CrayGNU/CrayGNU-19.06.eb failed (err: "build failed (first 300 chars): PrgEnv-gnu/6.0.5(76):ERROR:150: Module 'PrgEnv-gnu/6.0.5' conflicts with the currently loaded module(s) 'PrgEnv-cray/6.0.5'")
```

Reply:
Thanks @victorusu: I have amended `craytoolchain.py` for other reasons, since I was getting an error message on Tsa when unloading a non existing modulefile. It can be adapted to avoid this error on Piz Daint as well: if you want to go on in the meanwhile, just use `version: dummy`.

Reply:
Update for `CrayToolchain` to be included with EasyBuild v4.0.0 for this in https://github.com/easybuilders/easybuild-easyblocks/pull/1805 .

In EasyBuild v4.0, you can no longer use `dummy` as toolchain version to avoid loading the modules for the dependencies, so this change is a strict requirement now to avoid breaking Cray support...